tests/frontend/test_tdmprogram.py

```diff
@@ -452,3 +500,157 @@ class DummyCompiler(sf.compilers.compiler.Compiler):
 
     with pytest.raises(sf.program_utils.CircuitError, match=match):
         new_prog = prog.compile(device=spec, compiler=DummyCompiler())
+
+
+## Test for the compilation
+
+import blackbird as bb
+
+
+############################################################################
+# For the test below to work, the BB cript had to be changed
+# so that the squezing parameter is a dummy symbolic variable called p0
+# and now the names insides the BB script match the names in the
+# dictionary giving the allowed ranges for the variables
+# in the DeviceSpec object
+############################################################################
+
+tm = 4
+device_spec = {
+    "layout": "name template_tdm\nversion 1.0\ntarget tdm (shots=1)\ntype tdm (temporal_modes={tm}, copies=1)\nfloat array p0[1, {tm}] =\n    {{rs_array}}\nfloat array p1[1, {tm}] =\n    {{bs_array}}\nfloat array p2[1, {tm}] =\n    {{r_array}}\nfloat array p3[1, {tm}] =\n    {{m_array}}\n\nSgate(p0) | 1\nBSgate(p1) | (1, 0)\nRgate(p2) | 1\nMeasureHomodyne(p3) | 0\n",
+    "modes": {"concurrent": 2, "spatial": 1, "temporal": {"max": 100}},
+    "compiler": ["tdm"],
+    "gate_parameters": {
+        "p0": [0.5643],
+        "p1": [0, [0, 6.283185307179586]],
+        "p2": [0, [0, 3.141592653589793], 3.141592653589793],
+        "p3": [0, [0, 6.283185307179586]],
+    },
+}
+device_spec["layout"] = device_spec["layout"].format(tm=tm)
```

Comment:
Could we maybe have `{target}` inside of the layout and substitute that too explicitly `device_spec["layout"] = device_spec["layout"].format(target=target, tm=tm)`? Would conform with the original layout and it would not be missed when looking here.
```suggestion

target = "tdm"
tm = 4
device_spec = {
    "layout": "name template_tdm\nversion 1.0\ntarget {target} (shots=1)\ntype tdm (temporal_modes={tm}, copies=1)\nfloat array p0[1, {tm}] =\n    {{rs_array}}\nfloat array p1[1, {tm}] =\n    {{bs_array}}\nfloat array p2[1, {tm}] =\n    {{r_array}}\nfloat array p3[1, {tm}] =\n    {{m_array}}\n\nSgate(p0) | 1\nBSgate(p1) | (1, 0)\nRgate(p2) | 1\nMeasureHomodyne(p3) | 0\n",
    "modes": {"concurrent": 2, "spatial": 1, "temporal": {"max": 100}},
    "compiler": ["tdm"],
    "gate_parameters": {
        "p0": [0.5643],
        "p1": [0, [0, 6.283185307179586]],
        "p2": [0, [0, 3.141592653589793], 3.141592653589793],
        "p3": [0, [0, 6.283185307179586]],
    },
}
device_spec["layout"] = device_spec["layout"].format(target=target, tm=tm)
```